src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2318,13 +2320,46 @@ because xbuild doesn't support framework reference assemblies.
 	ResourceDirectories="$(MonoAndroidResDirIntermediate);@(LibraryResourceDirectories)"
 	ResourceNameCaseMap="$(_AndroidResourceNameCaseMap)"
   />
-  <Touch Files="$(IntermediateOutputPath)_javastubs.stamp" AlwaysCreate="True" />
+  <Touch Files="$(_AndroidStampDirectory)_GenerateJavaStubs.stamp" AlwaysCreate="True" />
+  <ItemGroup>
+    <FileWrites Include="$(IntermediateOutputPath)javastubs.cache" Condition="Exists('$(IntermediateOutputPath)javastubs.cache')" />
+  </ItemGroup>
+</Target>
+
+<Target Name="_ReadJavaStubsCache">
+  <ReadJavaStubsCache CacheFile="$(IntermediateOutputPath)javastubs.cache">
+    <Output TaskParameter="EmbeddedDSOsEnabled" PropertyName="_EmbeddedDSOsEnabled" />
+  </ReadJavaStubsCache>
+</Target>
+
+<Target Name="_SetupEmbeddedDSOs"
+    Condition=" '$(_EmbeddedDSOsEnabled)' == 'True' "
+    Inputs="$(IntermediateOutputPath)javastubs.cache"
+    Outputs="$(IntermediateOutputPath)dsoenvironment.txt">
+  <WriteLinesToFile
+      File="$(IntermediateOutputPath)dsoenvironment.txt"
+      Lines="__XA_DSO_IN_APK=1"
+      Overwrite="True"
+      WriteOnlyWhenDifferent="True"
```

Comment:
This feels counter to our documented MSBuild target guidelines. *When* this target is executed, then the `Outputs` *must* be updated so that the outputs *always* have a newer time stamp than the `Inputs`.

With `WriteOnlyWhenDifferent=“True”`, it feels like this invariant could be violated, in that `javastubs.cache` could have a newer timestamp *but no relevant changes* (e.g. someone `touch`ed it), and thus — as is — `dsoenvironment.txt` would *not* be “updated”/“touched” to have a newer timestamp than `javastubs.cache`.

Perhaps at minimum we need to remove this line?